src/adr/0051-email-rate-limiting.md

```diff
@@ -0,0 +1,40 @@
+# Email rate limiting
+
+Date: 2021-01-10
+
+## Status
+
+// todo
+
+## Context
+
+Rate limiting is one of the most popular features of any service. They limit how many emails you can send/receive from/to
+each email account over a given period of time.  
+We believe the rate-limiting will help James has more control of the resource and easy to dynamic config the user policy.
+
+## Decision
```

Comment:
We proposes to set up a new maven project dedicated to rate limiting. We proposes two implementation of rate limitings: in memroy (guava based) and redis. Please note that this will take the form of an extension jars, that could be dropped in ones James installation, and thus is optional, and not of a runtime dependency.